### DISCUSSION TITLE
Default values in related tables

### DISCUSSION THREAD

**Discussion by runningjustin** [babyfish-ct/jimmer#508] [Q&A]
Hey all - I have a `Users` entity with a `@OneToMany` relationship to an `Address` entity. Both entities have a String field called `status` with a `@Default` annotation. I've defined the DTO as follows:

```
input UserInput {
    #allScalars(Users)

    addresses @Serializable {
        #allScalars(Address)
        -lastUsedAt
        -status
        -useCount
    }
}
```

I specifically exclude `status` because it's not something I want exposed in the API. 
However, it seems that the `@Default` values in the `Address` relation don't get populated the same way the `@Default` values in the `Users` entity. Is it possible that the defaults for the embedded entity are not being populated?

**Comment by babyfish-ct** [comment]
`InputDTO` only controls the input shape of user. *(Outside logic, not the core of ORM)*

After the translation form InputDTO to entity, input will be forgot, so `@Default`/`DraftInterceptor` still works.*(Inside logic, core of ORM)*

They are two fully separated features.

**Reply by runningjustin** [reply]
Thanks for the response, however I don't think I was very clear in my original question. To put it succinctly, I have the following in my Address entity:

```
@Default("ACTIVE")
val status: String?
```
    
The status field is defined as not null in the database. Regardless of whether I exclude status from the DTO or not, it seems like this default is not being set if I don't supply the status field. There's no problem with a status field defined the same way in the 'outer' User entity. Not supplying this value always ends up generating a database constraint violation:
`org.postgresql.util.PSQLException: ERROR: null value in column "status" of relation "address" violates not-null constraint`

I am passing the DTO directly from the API call through to sqlClient:
`val savedRecord = DBUtils.sqlClient.entities.save(user)`

Thanks again...

**Comment by babyfish-ct** [comment]
For insert, it is right, but for update, ignoring it better, 

Please submit a FR: Ignore `@Default` for unloaded properties when update an object.

**Reply by babyfish-ct** [reply]
No, need not submit FR, it is the current logic.

0.8.122 release, all scalar properties support @Default
